zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,409 @@
+        if not task.slice_count:
+            task.slice_count = slice_count
+        elif task.slice_count != slice_count:
+            raise Exception(
+                "every upload request for image[uuid:%s] should has the same slice size and image size" % param.image_uuid)
+
+        return task
+
+    def upload_slice(self, entity, param, task):
+        # type: (Entity, UploadParam, UploadTask) -> None
+
+        boundary = get_boundary(entity)
```

Ruff (0.13.1)

Undefined name `get_boundary`

(F821)